### PR TITLE
Add `--memory-limit` PHP CLI option to the `aom:process` command instead of forcing 1GB memory_limit

### PR DESCRIPTION
**Pull request by KZeni**
I was running into an issue where the ./console aom:process command was requiring more than 1GB. I tried to change my own server's memory_limit via the php.ini, but that didn't do anything. I then found this script was hard-coding a memory_limit of 1GB. This isn't good as different people have different needs and it can certainly be an option rather than being hard-coded into a file which might be overwritten in the future by updating the version of the plugin.

In short, I made this default to the server's own memory_limit so those that instinctively go to modifying their php.ini won't be left confused. Also, I then added a `--memory-limit` option which then allows one to change the memory_limit value to whatever they might need (ex. `./console aom:process --memory-limit=2147483648` would give the script 2GB of memory).

### CODE REVIEW

**Comment by danielstonies** [comment]
Memory usage was improved within latest release. A --memory-limit is not needed any more for aom:process.

**Comment by KZeni** [comment]
I don't see what harm my PR would do if merged (even with the memory usage being improved.) I just checked and saw that the https://github.com/advanced-online-marketing/AOM/blob/master/Commands/EventProcessor.php file still has the memory limit hard-coded via `ini_set('memory_limit','1024M');`. I think there'd still be value in implementing what I've proposed here.
